server/pulp/server/content/web/views.py

```diff
@@ -51,52 +53,91 @@ def urljoin(scheme, host, port, base, path, query):
         url = [
             scheme,
             host,
-            port,
+            str(port),
             base,
             path,
             query
         ]
         return ''.join(url)
 
+    @staticmethod
+    def x_send(path):
+        """
+        Add the X-SENDFILE header to the returned reply causing Apache to send the file content.
+
+        :param path: The fully qualified *real* path to the requested content.
+        :type path: str
+        :return: An HTTP response.
+        :rtype: django.http.HttpResponse
+        """
+        if os.access(path, os.R_OK):
+            reply = HttpResponse()
+            reply['X-SENDFILE'] = path
+        else:
+            reply = HttpResponseForbidden()
+        return reply
+
     def __init__(self, **kwargs):
-        super(RedirectView, self).__init__(**kwargs)
+        super(ContentView, self).__init__(**kwargs)
         self.key = Key.load(pulp_conf.get('authentication', 'rsa_key'))
-        self.alias = AliasTable()
-        self.alias.load()
 
-    def get(self, request):
+    def redirect(self, request, path):
         """
         Redirected GET request.
 
         :param request: The WSGI request object.
         :type request: django.core.handlers.wsgi.WSGIRequest
-        :return: A redirect or not-found response.
+        :param path: The *real* path to stored content.
+        :type path: str
+        :return: A redirect or not-found reply.
         :rtype: django.http.HttpResponse
         """
-        try:
-            path = request.environ['REDIRECT_URL']
-        except KeyError:
-            # Not redirected by ErrorDocument
-            return HttpResponseNotFound()
+        enabled = parse_bool(pulp_conf.get('lazy', 'enabled'))
+
+        if not enabled:
+            # Not Redirected
+            return HttpResponseNotFound(path)
 
-        query = request.environ.get('REDIRECT_QUERY_STRING', '')
         scheme = request.environ['REQUEST_SCHEME']
         host = request.environ['SERVER_NAME']
         port = request.environ['SERVER_PORT']
-        enabled = parse_bool(pulp_conf.get('lazy', 'enabled'))
+        query = request.environ.get('REDIRECT_QUERY_STRING', '')
```

Comment:
Reading http://httpd.apache.org/docs/2.4/custom-error.html#variables makes me believe (though I have not actually tested it) that this variable will never be provided since we aren't using the ErrorDocument command.